packages/dappmanager/src/modules/installer/ensureEth2MigrationRequirements.ts

```diff
@@ -0,0 +1,33 @@
+import { InstallPackageData } from "../../types";
+import semver from "semver";
+import params from "../../params";
+import { listPackageNoThrow } from "../docker/list";
+
+/**
+ * Ensure the following requirements for Eth2 migration:
+ * - From dappmanager v0.2.46 prevent from installing legacy version (previous to remote signer support) for packages: (Prysm, Prysm-prater and Gnosis-Beacon-Chain-Prysm)
+ * -
+ */
+export async function ensureEth2MigrationRequirements(
+  packagesData: InstallPackageData[]
+): Promise<void> {
+  for (const pkg of packagesData) {
+    params.minimumAllowedPackageVersions.forEach(
```

Comment:
forEach cannot handle async functions, the thrown errors won't stop the installation

Reply:
Convert `params.minimumAllowedPackageVersions` into a `Map<string, string>` where key is dnpName, and value is minimum version. This way you don't have a quadratic loop